cocos/audio/apple/AudioPlayer.mm

```diff
@@ -122,6 +122,19 @@ of this software and associated documentation files (the "Software"), to deal
                 delete _rotateBufferThread;
                 _rotateBufferThread = nullptr;
                 ALOGVV("rotateBufferThread exited!");
+
+#if CC_TARGET_PLATFORM == CC_PLATFORM_IOS
+                // refer to: https://github.com/cocos2d/cocos2d-x/issues/18597
+                // some special OpenAL implementations existed on iOS platform, this can fix crash
+                ALint bufferProcessed = 0;
+                alGetSourcei(_alSource, AL_BUFFERS_PROCESSED, &bufferProcessed);
+                while (bufferProcessed < QUEUEBUFFER_NUM) {
+                    std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
Could this sleep waiting be smaller, it may lead to audio lag if there are buffers in queue.
Please test it with 1 or 2 milliseconds.